packages/askui-nodejs/src/utils/proxy/proxy-utils.ts

```diff
@@ -0,0 +1,111 @@
+import fs from 'fs';
+import http from 'http';
+import https from 'https';
+import { join } from 'path';
+import dns, { LookupOneOptions } from 'dns';
+
+const proxy = require('proxy');
+
+export const PROXY_HOSTNAME = 'proxy.proxy-unit-test-askui.com';
+export const SERVER_HOSTNAME = 'server.proxy-unit-test-askui.com';
+
+/** *
+ * This file is oriented by the integration tests of hpagent: https://github.com/delvedor/hpagent/blob/main/test/utils.js
+ */
+export function addBasicAuthentication(httpProxy: http.Server): http.Server {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, no-param-reassign
+  (httpProxy as any).authenticate = (req: any, fn: any) => {
+    const header = req.headers['proxy-authorization'] || ''; // get the auth header
+    const token = header.split(/\s+/).pop() || ''; // and the encoded auth token
+    const auth = Buffer.from(token, 'base64').toString(); // convert from base64
+    const parts = auth.split(/:/); // split on colon
+    const username = parts.shift(); // username is first
+    const password = parts.join(':');
+
+    if (!(username === 'username' && password === 'password')) {
+      fn(null, false);
+      return;
+    }
+    fn(null, true);
+  };
+  return httpProxy;
+}
+
+export async function buildProxy(port = 8009): Promise<http.Server> {
```

Comment:
magic value

Reply:
A random selected port number.

Reply:
Why not name it then this way `const RANDOMLY_SELECTED_PORT_NUMBER = 8009`? :D

Reply:
I switch this to the default proxy http port `3128` and https `3129`. ;-) This should be common knowledge. :stuck_out_tongue_winking_eye: